File: CSAdmin/LoginForm.vb
Public Class LoginForm

    Private Sub OK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK.Click
        'Login usando autentificación personalizada contra la BD
        Dim samplePrincipal As New SampleIPrincipal( _
            UsernameTextBox.Text, PasswordTextBox.Text)
        Me.PasswordTextBox.Text = ""
        If Not (samplePrincipal.Identity.IsAuthenticated) Then
            ' El usuario no esta validado.
            MessageBox.Show("Usuario o contraseña incorrecta, intente nuevamente.", _
                            "Error", MessageBoxButtons.OK, _
                            MessageBoxIcon.Information)
            PasswordTextBox.SelectAll()
            PasswordTextBox.Focus()
        Else
            ' Usuario Validado
            My.User.CurrentPrincipal = samplePrincipal
            Me.Close()
        End If
    End Sub

    Private Sub Cancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel.Click
        Me.Close()
        Application.Exit()
    End Sub

End Class

File: CSAdminApp/LoginForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using CSAdminApp.Clases;

namespace CSAdminApp
{
    public partial class LoginForm : Form
    {
        public LoginForm()
        {
            InitializeComponent();
            progressBar.Visible = false;
        }

        private void OK_Click(object sender, EventArgs e)
        {
            progressBar.Visible = true;
            try
            {
                SampleIPrincipal samplePrincipal = new SampleIPrincipal(this.UsernameTextBox.Text, this.PasswordTextBox.Text);
                this.PasswordTextBox.Clear();
                if (!samplePrincipal.Identity.IsAuthenticated)
                {
                    // Usuario No Validado
                    MessageBox.Show("Usuario o contraseña incorrecta, intente nuevamente.",
                            "Error", MessageBoxButtons.OK,
                            MessageBoxIcon.Information);
                    this.PasswordTextBox.Focus();
                }
                else
                {
                    // Usuario Validado, se actualiza CurrentPrincipal
                    System.Threading.Thread.CurrentPrincipal = samplePrincipal;
                    this.Close();
                }
                progressBar.Visible = false;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message + ex.InnerException.Message);
            }
        }

        private void Cancel_Click(object sender, EventArgs e)
        {
            this.Close();
            Application.Exit();
        }
    }
}
